### PR TITLE
Add flag for outer join and create nodes and visitors for equals expression

### PR DESCRIPTION
**Pull request by lasinicl**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/23172

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by grainier — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/model/tree/expressions/EqualsExpressionNode.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mohanvive) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/model/tree/expressions/EqualsExpressionNode.java:2]: Why license year is mentioned as 2017?

Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/model/tree/expressions/EqualsExpressionNode.java:2]: `EqualsExpressionNode.java` class is no longer available.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangEqualsExpr.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangEqualsExpr.java:2]: License year 2018?

File: tests/jballerina-unit-test/src/test/resources/test-src/query/join-clause.bal [hunk 3]
```diff
@@ -73,20 +73,43 @@ function testJoinClauseWithStream() returns boolean {
...
+       on person.id == 1
+       select {
+           fname : person.fname,
+           lname : person.lname,
+           dept : "Eng"
+       };
+
+    return deptPersonList;
+}
+
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/query/join-clause.bal:114]: unnecessary new line

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangEqualsExpr.java (hunk 2 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangEqualsExpr.java:2]: BLangEqualsExpr.java class is no longer available.

File: tests/jballerina-unit-test/src/test/resources/test-src/query/join-clause.bal (hunk 3 above)
Comment (lasinicl) [reply tests/jballerina-unit-test/src/test/resources/test-src/query/join-clause.bal:114]: Already fixed

**Pull request merged**